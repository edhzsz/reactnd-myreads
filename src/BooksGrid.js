import React from 'react'
import Book from './Book.js'

const BooksGrid = ( { books } ) => {
  return (
      <ol className="books-grid">
        {books && books.map(book => (
          <li key={book.id}>
            <Book book={book}  />
          </li>
          )
        )}
      </ol>
    );
}

export default BooksGrid;