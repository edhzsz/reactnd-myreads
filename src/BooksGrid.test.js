import React from 'react';
import ReactDOM from 'react-dom';
import BooksGrid from './BooksGrid.js';
import renderer from 'react-test-renderer';

describe('Component: BooksGrid', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BooksGrid />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <BooksGrid />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});