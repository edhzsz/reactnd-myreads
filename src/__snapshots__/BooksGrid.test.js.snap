// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BooksGrid should match its empty snapshot 1`] = `
<ol
  className="books-grid"
/>
`;

exports[`Component: BooksGrid should render the books array 1`] = `
<ol
  className="books-grid"
>
  <li>
    <Book
      book={
        Object {
          "authors": Array [
            "Harper Lee",
          ],
          "coverUrl": "http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api",
          "id": "nggnmAEACAAJ",
          "shelf": "currentlyReading",
          "title": "To Kill a Mockingbird",
        }
      }
    />
  </li>
  <li>
    <Book
      book={
        Object {
          "authors": Array [
            "Orson Scott Card",
          ],
          "coverUrl": "http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api",
          "id": "sJf1vQAACAAJ",
          "shelf": "currentlyReading",
          "title": "Ender's Game",
        }
      }
    />
  </li>
</ol>
`;
